templates/stamp/mongodb-ubuntu-install.sh

```diff
@@ -0,0 +1,413 @@
+PACKAGE_NAME=mongodb-org
+REPLICA_SET_KEY_DATA=""
+REPLICA_SET_NAME=""
+REPLICA_SET_KEY_FILE="/etc/mongo-replicaset-key"
+DATA_DISKS="/datadisks"
+DATA_MOUNTPOINT="$DATA_DISKS/disk1"
+MONGODB_DATA="$DATA_MOUNTPOINT/mongodb"
+MONGODB_DATA_INTERIM_ROOTPATH="/mongo"
+MONGODB_PORT=27017
+IS_ARBITER=false
+IS_LAST_MEMBER=false
+JOURNAL_ENABLED=true
+ADMIN_USER_NAME=""
+ADMIN_USER_PASSWORD=""
+INSTANCE_COUNT=1
+NODE_IP_PREFIX="10.0.0.1"
+LOGGING_KEY="[logging-key]"
+OFFSET=0
+
+help()
+{
+	echo "This script installs MongoDB on the Ubuntu virtual machine image"
+	echo "Options:"
+	echo "		-i Installation package URL"
+	echo "		-b Installation package name"
+	echo "		-r Replica set name"
+	echo "		-k Replica set key"
+	echo "		-u System administrator's user name"
+	echo "		-p System administrator's password"
+	echo "		-x Member node IP prefix"	
+	echo "		-n Number of member nodes"	
+	echo "		-a (arbiter indicator)"	
+	echo "		-l (last member indicator)"	
+	echo "		-o (IP Address Offset)"	
+}
+
+log()
+{
+	# If you want to enable this logging add a un-comment the line below and add your account key 
+	#curl -X POST -H "content-type:text/plain" --data-binary "$(date) | ${HOSTNAME} | $1" https://logs-01.loggly.com/inputs/${LOGGING_KEY}/tag/redis-extension,${HOSTNAME}
+	echo "$1"
+}
+
+log "Begin execution of MongoDB installation script extension on ${HOSTNAME}"
+
+if [ "${UID}" -ne 0 ];
+then
+    log "Script executed without root permissions"
+    echo "You must be root to run this program." >&2
+    exit 3
+fi
+
+# Parse script parameters
+while getopts :i:b:r:k:u:p:x:n:o:alh optname; do
+
+	# Log input parameters (except the admin password) to facilitate troubleshooting
+	if [ ! "$optname" == "p" ] && [ ! "$optname" == "k" ]; then
+		log "Option $optname set with value ${OPTARG}"
+	fi
+  
+	case $optname in
+	i) # Installation package location
+		PACKAGE_URL=${OPTARG}
+		;;
+	b) # Installation package name
+		PACKAGE_NAME=${OPTARG}
+		;;
+	r) # Replica set name
+		REPLICA_SET_NAME=${OPTARG}
+		;;	
+	k) # Replica set key
+		REPLICA_SET_KEY_DATA=${OPTARG}
+		;;	
+	u) # Administrator's user name
+		ADMIN_USER_NAME=${OPTARG}
+		;;		
+	p) # Administrator's user name
+		ADMIN_USER_PASSWORD=${OPTARG}
+		;;	
+	x) # Private IP address prefix
+		NODE_IP_PREFIX=${OPTARG}
+		;;				
+	n) # Number of instances
+		INSTANCE_COUNT=${OPTARG}
+		;;		
+	o) # IP Address Offset
+		OFFSET=${OPTARG}
+		;;		
+	a) # Arbiter indicator
+		IS_ARBITER=true
+		JOURNAL_ENABLED=false
+		;;		
+	l) # Last member indicator
+		IS_LAST_MEMBER=true
+		;;		
+    h)  # Helpful hints
+		help
+		exit 2
+		;;
+    \?) # Unrecognized option - show help
+		echo -e \\n"Option -${BOLD}$OPTARG${NORM} not allowed."
+		help
+		exit 2
+		;;
+  esac
+done
+
+# Validate parameters
+if [ "$ADMIN_USER_NAME" == "" ] || [ "$ADMIN_USER_PASSWORD" == "" ];
+then
+    log "Script executed without admin credentials"
+    echo "You must provide a name and password for the system administrator." >&2
```

Comment:
It's a very fine line. Some echo statements are targeted at stderrorn and others are used to write values. Using log() for those isn't appropriate. By default log messages go to stdout. We can take up more elaborate logging in a subsequent pass.